module_utils/network_lsr/argument_validator.py

```diff
@@ -1260,6 +1279,45 @@ def _validate_post(self, value, name, result):
         return result
 
 
+class ArgValidatorListMatch(ArgValidatorList):
+    def __init__(self, name, nested, default_value, remove_none_or_empty):
+        ArgValidatorList.__init__(
+            self,
+            name,
+            nested,
+            default_value,
+            remove_none_or_empty,
+        )
+
+    def _validate_impl(self, value, name):
+        result = ArgValidatorList._validate_impl(self, value, name)
+        if result == ["|"] or result == ["&"]:
+            raise ValidationError(
+                name,
+                "value '%s' is not a valid 'match.path' setting, after "
+                "normalization, '%s' will only match the devices that have no PCI "
+                "path" % (value, result),
+            )
+        return result
```

Comment:
The feature is working well, IMO. I tried it out and it works as described in `README.md`. I've been discussing this part of the code with @tyll. This can be problematic in the future because the validator is supposed to be generic but the error and the checking are for 'match.patch'.

I have two possible solutions:
1. Implement now a specific validator for match.path based on `ArgValidatorListMatch` and include a test to check we are triggering the error.
2. Put a comment here with a `TODO:` in case we implement a different match setting in the future this will need to be changed.

What do you think? @liangwen12year @tyll

Reply:
If we need to implement another match setting in the future, we can just introduce different flags (e.g. `is_match_path`, `is_match_name`) to the param list of ArgvalidatorListMatch, and we can use those flags to control how we want to validate in the `_validate_impl()`:

```python
def _validate_impl(self, value, name):
    if is_match_path: 
        result = ArgValidatorList._validate_impl(self, value, name)
        if result == ["|"] or result == ["&"]:
            raise ValidationError(
                name,
                "value '%s' is not a valid 'match.path' setting, after "
                "normalization, '%s' will only match the devices that have no PCI "
                "path" % (value, result),
            )
        return result
   
    if is_match_name:
      ...

```

Reply:
I didn't see it earlier, let's call the current class ..ListMatchPath then it is clear that it is specific and a better solution can be added when we need to support other match settings.

Reply:
I renamed class `ArgValidatorListMatch` into `ArgValidatorListMatchPath`.